xtrack/beam_elements/elements.py

```diff
@@ -84,6 +84,171 @@ def add_slice(weight, container, thick_name, slice_name, _buffer=None):
+        x^n*y^m that represent the nonlinearities of the map in the horizontal plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fx_y_exps: array of floats
+        An array containing the exponents, m, of y for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the horizontal plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fy_coeffs: array of floats
+        An array that contains the coefficients of monomials of the form x^n*y*m that 
+        represent the nonlinearities of the map in the vertical plane. It is 
+        calculated at initialisation based on the multipole coefficients provided.
+    fy_x_exps: array of floats
+        An array containing the exponents, n, of x for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the vertical plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fy_y_exps: array of floats
+        An array containing the exponents, m, of y for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the vertical plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    n_fx_coeffs: int
+        Length of the arrays fx_coeffs, fx_x_exps, and fx_y_exps.
+    n_fy_coeffs: int
+        Length of the arrays fy_coeffs, fy_x_exps, and fy_y_exps.
+    n_turns: int
+        Number of turns to track. Default is ``1``.
+    norm: int
+        1 if input coordinates are already normalised, 0 if not. Default is ``0``.
+
+    Comments
+    --------
+
+    - The user provides omega_x and omega_y, their sin and cos is calculated at 
+      initialisation.
+    - The user provides "multipole_coeffs", an array of floats, that contains the 
+      strength of the multipoles present in the map. Default is ``[0]``.
+
+    '''
+
+    _xofields = {
+        'sin_omega_x': xo.Float64,
+        'cos_omega_x': xo.Float64,
+        'sin_omega_y': xo.Float64,
+        'cos_omega_y': xo.Float64,
+        'n_turns': xo.Int64,
+        'twiss_params': xo.Float64[:],
+        'domegax': xo.Float64,
+        'domegay': xo.Float64,
+        'dx': xo.Float64,
+        'ddx': xo.Float64,
+        'fx_coeffs': xo.Float64[:],
+        'fx_x_exps': xo.Int64[:],
+        'fx_y_exps': xo.Int64[:],
+        'fy_coeffs': xo.Float64[:],
+        'fy_x_exps': xo.Int64[:],
+        'fy_y_exps': xo.Int64[:],
+        'n_fx_coeffs': xo.Int64,
+        'n_fy_coeffs': xo.Int64,
+        'norm': xo.Int64,
+    }
+
+    isthick = False
+    behaves_like_drift = False
+
+    _extra_c_sources = [_pkg_root.joinpath('beam_elements/elements_src/henonmap.h')]
+
+    def __init__(self, omega_x = 0.,
+                       omega_y = 0.,
+                       n_turns = 1, 
+                       twiss_params = [0., 1., 0., 1.],
```

Comment:
Using lists as default arguments in Python should be avoided (https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments), as lists are mutable.

This won't cause problems in your particular case, but for good practice consider using a tuple instead, or `None` and a check for `is None` in the body of the function.

Reply:
I changed the default to None and throw a value error if nothing is provided.